src/openrct2/scripting/ScNetwork.hpp

```diff
@@ -0,0 +1,377 @@
+#    endif
+        }
+
+        std::vector<std::string> permissions_get()
+        {
+#    ifndef DISABLE_NETWORK
+            auto index = network_get_group_index(_id);
+            if (index == -1)
+                return {};
+
+            // Create array of permissions
+            std::vector<std::string> result;
+            auto permissionIndex = 0;
+            for (const auto& action : NetworkActions::Actions)
+            {
+                if (network_can_perform_action(index, permissionIndex))
+                {
+                    auto p = TransformPermissionKeyToJS(action.PermissionName);
+                    result.push_back(p);
+                }
+                permissionIndex++;
+            }
+            return result;
+#    else
+            return {};
+#    endif
+        }
+        void permissions_set(std::vector<std::string> value)
+        {
+#    ifndef DISABLE_NETWORK
+            auto groupIndex = network_get_group_index(_id);
+            if (groupIndex == -1)
+                return;
+
+            // First clear all permissions
+            auto networkAction = NetworkModifyGroupAction(
+                ModifyGroupType::SetPermissions, _id, "", 0, PermissionState::ClearAll);
+            GameActions::Execute(&networkAction);
+
+            std::vector<bool> enabledPermissions;
+            enabledPermissions.resize(NetworkActions::Actions.size());
+            for (const auto& p : value)
+            {
+                auto permissionName = TransformPermissionKeyToInternal(p);
+
+                auto permissionIndex = 0;
+                for (const auto& action : NetworkActions::Actions)
+                {
+                    if (action.PermissionName == permissionName)
+                    {
+                        enabledPermissions[permissionIndex] = true;
+                    }
+                    permissionIndex++;
+                }
+            }
+
+            for (size_t i = 0; i < enabledPermissions.size(); i++)
+            {
+                auto toggle = (enabledPermissions[i] != (network_can_perform_action(groupIndex, (uint32_t)i) != 0));
+                if (toggle)
+                {
+                    auto networkAction2 = NetworkModifyGroupAction(
+                        ModifyGroupType::SetPermissions, _id, "", (uint32_t)i, PermissionState::Toggle);
+                    GameActions::Execute(&networkAction2);
+                }
+            }
+#    endif
+        }
+
+        static void Register(duk_context* ctx)
+        {
+            dukglue_register_property(ctx, &ScPlayerGroup::id_get, nullptr, "id");
+            dukglue_register_property(ctx, &ScPlayerGroup::name_get, &ScPlayerGroup::name_set, "name");
+            dukglue_register_property(ctx, &ScPlayerGroup::permissions_get, &ScPlayerGroup::permissions_set, "permissions");
+        }
+
+    private:
+        std::string TransformPermissionKeyToJS(const std::string& s)
+        {
+            auto result = s.substr(11);
```

Comment:
Just to make it more clear:
```suggestion
            auto result = s.substr(sizeof("PERMISSION_") - 1);
```